### DISCUSSION TITLE
Github acc lock?

### DISCUSSION THREAD

**Discussion by dedfft** [coder/code-server#4147] [Q&A]
Hello dear community. Didn’t find it in doc. After first login, is my code server locked to my only github account? Or anyone can login? How can i limit that?

**Comment by lg-nineoneninethreenine** [comment]
I didn't understand. As you told,


> After first login, is my code server locked to my only github account? Or anyone can login? How can i limit that?

I don't think so. After first login, you will have to set up code-server and connect your GitHub account. Unless you use `code-server --link`(proxies code-server, produces a domain that you can use to access it from anywhere with your GitHub account) and incidentally, your GitHub account got hacked or somebody who you know is using your GitHub account, they can access code-server via the link. 

But I think the viable answer is that code-server is not locked to your account, that is, you can add any GitHub accounts. Anyone can login if they know your code-server password(keep that private or that's it) or if you use `code-server --link` then they can use your GitHub account and log in. But you can set a limit, it has to be in the docs.

**Comment by dedfft** [comment]
Thank you for answer!
I used this install script to launch it, so i guess it using link to authorize anyone with github account?
How can i limit that, because I didn’t setup a password

**Reply by lg-nineoneninethreenine** [reply]
No, the `--link` tag gives you a URL that can unlock code-server only via your GitHub handle. On first setup, it will produce a GitHub OAuth page and you have to authorise. Next, on the terminal, it will give you a link. Whenever you access it from another computer, you will have to log in to GitHub first. It removes the password part and adds TLS, Authentication via GitHub, port forwarding, and a special link for you. But note this that it needs latency below 75 ms. This is applicable for cloud VMs, like Google Cloud, AWS, Azure, etc. For you, you must change the password. Open your favourite text editor, and open this file: `~/.config/code-server/config.yaml`. Then, change the password in the password part of the config. Don't change anything else.

> so i guess it using link to authorize anyone with github account?

No, it authorizes only your GitHub account.

**Reply by lg-nineoneninethreenine** [reply]
Use normal code-server. There are issues with the `--link` tag, since it's in beta. But I suggest that you use `--link` on a cloud VM or if you have an ultra-fast internet connection, try that.

**Reply by dedfft** [reply]
I use it on digital ocean. Didn’t see a problem, works even on ipad, i am just worried  that anyone with github acc can authorize as easy as i do via server ip address, because it redirects to github ouauth immediately in private mode. 

I wish i used password, but as i told, it happened very fast with digital ocean droplet and install script. 

For the sake of test and own calm, i will try login with different github)

**Reply by lg-nineoneninethreenine** [reply]
Ah, just use `--link`! On first trial, as I told, it will produce a link on the console for the GitHub OAuth. After allowing that, you get the link. As I told too, you get full protection. I also need to tell(I forgot to mention this) that you are safe, since even if the hacker guys get your IP, they can't guess the port since code-server's port is randomized on every `--link` launch. And NOBODY will log in, since they need your account. You can shut code-server down anytime too! This is important. After your use of code-server, shut the VM down. It will close all programs and hacker guys won't access it at all.

**Reply by dedfft** [reply]
Thank you so much! Works perfectly and secure. Just checked with new acc and it was FORBIDDEN)) Very happy now)
